dns.go

```diff
@@ -133,17 +133,20 @@ func (api *API) DNSRecord(ctx context.Context, zoneID, recordID string) (DNSReco
 //
 // API reference: https://api.cloudflare.com/#dns-records-for-a-zone-update-dns-record
 func (api *API) UpdateDNSRecord(ctx context.Context, zoneID, recordID string, rr DNSRecord) error {
-	rec, err := api.DNSRecord(ctx, zoneID, recordID)
```

Comment:
thanks, this looks like a hangover from a80236a02adfc1fc95be394d3bc5ddf1501caa34 where we swapped from HTTP PUT => PATCH but still wanted the option for end users to only supply a single attribute.

Reply:
Got it. Is there something else I should change in this PR?

Reply:
Not at this stage. I'm in the process of reviewing and checking this as it's pretty critical.